sktree/tree/unsupervised/_unsup_criterion.pyx

```diff
@@ -398,3 +396,159 @@ cdef class TwoMeans(UnsupervisedCriterion):
         self.n_node_samples = end - start
         self.start = start
         self.end = end
+
+cdef class FastBIC(UnsupervisedCriterion):
+    """Placeholder for FastBIC docs
+    """
+
+    cdef double sum_of_squares(
+        self,
+        SIZE_t start,
+        SIZE_t end,
+        double mean,
+    ) nogil:
+        """Computes variance of feature vector from sample_indices[start:end].
+
+        See: https://en.wikipedia.org/wiki/Weighted_arithmetic_mean#Weighted_sample_variance.  # noqa
+
+        Parameters
+        ----------
+        start : SIZE_t
+            The start pointer
+        end : SIZE_t
+            The end pointer.
+        mean : double
+            The precomputed mean.
+
+        Returns
+        -------
+        ss : double
+            Sum of squares
+        """
+        cdef SIZE_t s_idx, p_idx        # initialize sample and pointer index
+        cdef double ss = 0.0            # sum-of-squares
+        cdef DOUBLE_t w = 1.0           # optional weight
+
+        # calculate variance for the sample_indices chosen start:end
+        for p_idx in range(start, end):
+            s_idx = self.sample_indices[p_idx]
+
+            # include optional weighted sum of squares
+            if self.sample_weight is not None:
+                w = self.sample_weight[s_idx]
+
+            ss += w * (self.Xf[s_idx] - mean) * (self.Xf[s_idx] - mean)
+        return ss
+
+    cdef double node_impurity(
+        self
+    ) nogil:
+        """Evaluate the impurity of the current node.
+
+        TODO: describe FastBIC
+
+        1. Prior = w = n/N
+        2. Mean = u
+        3. Vars = sig
+
+        impurity = -2*n*log(w) + n*log(2*pi*sig)
+        """
+        cdef double mean
+        cdef double impurity
+        cdef SIZE_t n_samples = self.n_samples #TODO: define n_samples
+        cdef SIZE_t n_node_samples = self.n_node_samples
+
+        # If calling without setting the
+        if self.Xf is None:
+            with gil:
+                raise MemoryError(
+                    'Xf has not been set yet, so one must call init_feature_vec.'
+                )
+
+        # first compute mean
+        mean = self.sum_total / n_node_samples
```

Comment:
Should probably normalize with `weighted_n_node_samples` instead of `n_node_samples`. Can you also fix this inside `TwoMeans.node_impurity`?

Reply:
Thank you! I just fixed both instances